In [27]:
# importing necessary libraries
import pandas as pd

In [28]:
# Load the dataset
df = pd.read_csv("../data/DSI_kickstarterscrape_dataset.csv", encoding="latin1")

In [29]:
# Get the first 5 rows of the dataset
df.head()

,project id,name,url,category,subcategory,location,status,goal,pledged,funded percentage,backers,funded date,levels,reward levels,updates,comments,duration
0,39409,WHILE THE TREES SLEEP,http://www.kickstarter.com/projects/emiliesaba...,Film & Video,Short Film,"Columbia, MO",successful,10500.0,11545.0,1.099524,66,"Fri, 19 Aug 2011 19:28:17 -0000",7,"$25,$50,$100,$250,$500,$1,000,$2,500",10,2,30.00
1,126581,Educational Online Trading Card Game,http://www.kickstarter.com/projects/972789543/...,Games,Board & Card Games,"Maplewood, NJ",failed,4000.0,20.0,0.005000,2,"Mon, 02 Aug 2010 03:59:00 -0000",5,"$1,$5,$10,$25,$50",6,0,47.18
2,138119,STRUM,http://www.kickstarter.com/projects/185476022/...,Film & Video,Animation,"Los Angeles, CA",live,20000.0,56.0,0.002800,3,"Fri, 08 Jun 2012 00:00:31 -0000",10,"$1,$10,$25,$40,$50,$100,$250,$1,000,$1,337,$9,001",1,0,28.00
3,237090,GETTING OVER - One son's search to finally kno...,http://www.kickstarter.com/projects/charnick/g...,Film & Video,Documentary,"Los Angeles, CA",successful,6000.0,6535.0,1.089167,100,"Sun, 08 Apr 2012 02:14:00 -0000",13,"$1,$10,$25,$30,$50,$75,$85,$100,$110,$250,$500...",4,0,32.22
4,246101,The Launch of FlyeGrlRoyalty &quot;The New Nam...,http://www.kickstarter.com/projects/flyegrlroy...,Fashion,Fashion,"Novi, MI",failed,3500.0,0.0,0.000000,0,"Wed, 01 Jun 2011 15:25:39 -0000",6,"$10,$25,$50,$100,$150,$250",2,0,30.00


In [30]:
# Get the dimension of the dataset
df.shape

(45957, 17)

In [31]:
# Get the summary of the dataset
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 45957 entries, 0 to 45956
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   project id         45957 non-null  int64  
 1   name               45957 non-null  str    
 2   url                45957 non-null  str    
 3   category           45957 non-null  str    
 4   subcategory        45957 non-null  str    
 5   location           44635 non-null  str    
 6   status             45957 non-null  str    
 7   goal               45957 non-null  float64
 8   pledged            45945 non-null  float64
 9   funded percentage  45957 non-null  float64
 10  backers            45957 non-null  int64  
 11  funded date        45957 non-null  str    
 12  levels             45957 non-null  int64  
 13  reward levels      45898 non-null  str    
 14  updates            45957 non-null  int64  
 15  comments           45957 non-null  int64  
 16  duration           45957 non-null

In [32]:
# Check for missing values in the dataset
missing_count = df.isnull().sum()
missing_count[missing_count > 0]

location         1322
pledged            12
reward levels      59
dtype: int64

In [33]:
# Calculate the percentage of missing values for each column to identify which columns have a significant amount of missing data
missing_percentage = (missing_count / len(df)) * 100
missing_percentage[missing_percentage > 0].round(2)

location         2.88
pledged          0.03
reward levels    0.13
dtype: float64

In [34]:
# Check the first 10 rows with missing values in the "location" column
df[df["location"].isnull()].head(10)

,project id,name,url,category,subcategory,location,status,goal,pledged,funded percentage,backers,funded date,levels,reward levels,updates,comments,duration
11,727286,Offline Wikipedia iPhone app,http://www.kickstarter.com/projects/dphiffer/o...,Technology,Open Software,NaN,successful,99.0,145.0,1.464646,25,"Tue, 14 Jul 2009 06:59:59 -0000",1,$1,5,19,79.64
14,893085,Esperanza Farm: A Novel,http://www.kickstarter.com/projects/JesusMaria...,Publishing,Fiction,NaN,failed,6500.0,765.0,0.117692,20,"Fri, 16 Jul 2010 03:59:00 -0000",7,"$5,$10,$25,$50,$75,$150,$300",10,0,60.34
49,2442649,Dream with Me--A documentary about one year in...,http://www.kickstarter.com/projects/1800556280...,Film & Video,Documentary,NaN,successful,5000.0,5615.0,1.123000,90,"Fri, 16 Oct 2009 01:08:00 -0000",8,"$10,$25,$50,$100,$500,$1,000,$2,500,$5,000",6,7,30.89
98,4719237,Indie Music Website (iTunes for Indie Musicians),http://www.kickstarter.com/projects/binarykoll...,Music,Indie Rock,NaN,failed,10000.0,10.0,0.001000,1,"Thu, 01 Oct 2009 06:19:00 -0000",3,"$1,$20,$100",0,0,29.98
100,4732285,Insiders/Out: Exploring Outsider Art in America,http://www.kickstarter.com/projects/insidersou...,Art,Art,NaN,successful,2000.0,2000.0,0.999830,30,"Thu, 08 Jul 2010 21:39:00 -0000",13,"$1,$5,$10,$15,$20,$25,$35,$50,$75,$100,$150,$2...",4,1,42.97
104,4948516,STRANGE POSITIONING SYSTEMS (SPS) - A global l...,http://www.kickstarter.com/projects/sps/strang...,Art,Performance Art,NaN,failed,8500.0,695.0,0.081765,15,"Fri, 20 Aug 2010 21:37:00 -0000",10,"$10,$25,$50,$75,$100,$250,$500,$600,$1,000,$1,500",3,1,42.19
120,5785733,Faith - a film by Eli Daughdrill,http://www.kickstarter.com/projects/771183968/...,Film & Video,Film &amp; Video,NaN,failed,50000.0,7890.0,0.157800,25,"Fri, 16 Apr 2010 05:48:00 -0000",8,"$10,$25,$50,$100,$250,$500,$1,000,$5,000",8,2,74.39
151,7808555,Fabric Fiction: A collaborative literary exper...,http://www.kickstarter.com/projects/1930061782...,Publishing,Fiction,NaN,failed,3000.0,582.0,0.194000,13,"Fri, 04 Jun 2010 16:30:00 -0000",10,"$1,$5,$10,$12,$20,$25,$40,$50,$75,$100",7,0,89.94
176,9094253,"The People, Places &amp; Patterns Project",http://www.kickstarter.com/projects/pwc/the-pe...,Photography,Photography,NaN,failed,10000.0,3078.0,0.307752,43,"Tue, 16 Feb 2010 19:11:00 -0000",9,"$30,$60,$80,$100,$150,$250,$500,$1,000,$2,000",11,6,61.67
186,9316432,Produce JigGsaw's never before heard &quot;The...,http://www.kickstarter.com/projects/jiggsaw/pr...,Music,Rock,NaN,failed,700.0,25.0,0.035714,1,"Mon, 07 Jun 2010 04:00:00 -0000",6,"$15,$25,$50,$100,$500,$1,000",0,0,89.78


In [35]:
# Verify that the status of the campaigns with missing locations is either successful or failed
df[df["location"].isnull()]["status"].value_counts()

status
failed        692
successful    630
Name: count, dtype: int64

Location has 1,322 missing values (2.88%). The affected records include both successful and failed campaigns and still contain useful campaign data. Instead of dropping these rows, missing locations will be labeled as "Unknown" to preserve the records without assuming a location.

In [36]:
# Fill missing locations with "Unknown"
df["location"] = df["location"].fillna("Unknown")

In [37]:
# Verify there are no missing locations left
df["location"].isnull().sum()

np.int64(0)

In [38]:
# Inspect missing pledged values to see whether related campaign fields reveal a pattern
df[df["pledged"].isnull()][["name", "goal", "pledged", "funded percentage", "status"]]

,name,goal,pledged,funded percentage,status
1187,Xenonauts,50000.0,NaN,2.219487,live
4502,Twokinds Book Printing Drive,25000.0,NaN,6.303783,live
13381,HICKIES - TURN YOUR KICKS INTO SLIP-ONS,25000.0,NaN,4.907843,live
13802,Genie - Motion control time lapse device,150000.0,NaN,2.999618,live
25239,B9Creator - A High Resolution 3D Printer,50000.0,NaN,4.761380,live
29412,Phil Tippett's &quot;MAD GOD&quot;,40000.0,NaN,2.532484,live
31164,gTar: The First Guitar That Anybody Can Play,100000.0,NaN,2.846420,live
34274,Space Command,75000.0,NaN,1.796187,live
35032,BronyCon: The Documentary,60000.0,NaN,3.148185,live
40759,Two Guys SpaceVenture - by the creators of Spa...,500000.0,NaN,0.583738,live


### Pledged

There are 12 missing pledged values (0.03% of the dataset).
All 12 occur in campaigns with `status = "live"`.

All 12 missing values occur in campaigns with `status = "live"`. Task #2 filters the dataset to retain only `successful` and `failed` campaigns for the binary classification objective, so these `live` campaigns are removed during preprocessing.

Therefore, no imputation is needed for the missing `pledged` values.

In [39]:
# Check whether funded percentage is approximately pledged / goal
df[["goal", "pledged", "funded percentage"]].dropna().head(10)

,goal,pledged,funded percentage
0,10500.0,11545.0,1.099524
1,4000.0,20.0,0.005000
2,20000.0,56.0,0.002800
3,6000.0,6535.0,1.089167
4,3500.0,0.0,0.000000
5,3500.0,3582.0,1.023331
6,1000.0,280.0,0.280000
7,2000.0,2180.0,1.090000
8,1000.0,1125.0,1.125000
9,7500.0,9836.0,1.311527


In [40]:
# Inspect missing reward levels values to see whether related campaign fields reveal a pattern
df[df["reward levels"].isnull()][["name", "levels", "reward levels", "status"]]

,name,levels,reward levels,status
78,DEAR HARVEY (Stories of Harvey Milk)...Educati...,0,NaN,failed
830,Having homeless create public art,0,NaN,failed
2720,The FullCircle Project: Skiing and Service in ...,0,NaN,successful
3067,Gifts and Crafts made from wood(Golf Cart),0,NaN,failed
3889,The Pyrosphere: large scale computer controlle...,0,NaN,failed
5129,Reanimation Library Coffer Builder,0,NaN,successful
8591,ENUR REQUIEM is going back into the studio!!!!!,0,NaN,failed
10624,Back &quot;The Pack&quot;!!!,0,NaN,successful
10712,Ballard High School Debate,0,NaN,successful
10749,THE FUTURE,0,NaN,failed


### Reward Levels

There are 59 missing values in `reward levels` (0.13% of the dataset).

All 59 missing values occur where `levels = 0`, which indicates these campaigns have no reward tiers rather than unknown reward information.

Therefore, the missing values will be replaced with `"No reward levels"`.

In [41]:
# Fill missing reward levels with "No reward levels"
df["reward levels"] = df["reward levels"].fillna("No reward levels")

In [42]:
# Verify there are no missing reward levels left
df["reward levels"].isnull().sum()

np.int64(0)

In [43]:
# Check for any remaining missing values in the dataset
remaining_missing = df.isnull().sum()
remaining_missing[remaining_missing > 0]

pledged    12
dtype: int64

## Missing Values Summary

The dataset contains 45,957 rows and 17 columns. Missing values were identified in three columns:

| Column | Missing Count | Missing % | Strategy |
|---|---:|---:|---|
| location | 1,322 | 2.88% | Replaced with `"Unknown"` |
| pledged | 12 | 0.03% | No imputation needed; rows are removed during outcome filtering |
| reward levels | 59 | 0.13% | Replaced with `"No reward levels"` |

### Decisions

- **Location:** The missing-location records contained valid campaign information and included both successful and failed campaigns. Rather than dropping 2.88% of the dataset, missing locations were labeled `"Unknown"`.

- **Pledged:** In the original dataset, all 12 missing `pledged` values belonged to `live` campaigns. These campaigns were removed during Task #2 when the dataset was filtered to retain only `successful` and `failed` outcomes, so no imputation was required.

- **Reward Levels:** All 59 missing values occurred where `levels = 0`, indicating that these campaigns had no reward tiers. These values were replaced with `"No reward levels"`.